### PR TITLE
Support fully qualified git clone uris for 'basher install'

### PR DESCRIPTION
**Pull request by lordvlad**
First shot at closing #67

### CODE REVIEW

File: tests/basher-_clone.bats [hunk 1]
```diff
@@ -77,5 +77,13 @@ load test_helper
...
-  assert_output "git clone --depth=1 --recursive git@site:username/package.git ${BASHER_PACKAGES_PATH}/username/package"
+  assert_output "git clone --depth=1 --recursive ssh://git@site/username/package.git ${BASHER_PACKAGES_PATH}/username/package"
+}
+
+@test "using full clone uri" {
+  mock_command git
+
+  run basher-_clone true site group/subgroup/subsubgroup/package
+  assert_success
+  assert_output "git clone --depth=1 --recursive ssh://git@site/group/subgroup/subsubgroup/package.git ${BASHER_PACKAGES_PATH}/group/subgroup/subsubgroup/package"
```
Comment (juanibiapina) [inline tests/basher-_clone.bats]: Nice that this works out of the box.

File: libexec/basher-install [hunk 2]
```diff
@@ -21,8 +21,28 @@ if [ "$#" -ne 1 ]; then
 fi
 
 if [[ "$1" = */*/* ]]; then
-  IFS=/ read -r site user name <<< "$1"
-  package="${user}/${name}"
+  # Handle fully qualified urls which start with a protocol.
+  if [[ "$1" = *//* ]]; then
```
Comment (juanibiapina) [inline libexec/basher-install]: Maybe using `*://*` here is a better idea for detecting the protocol?

**Comment by juanibiapina** [comment]
Overall I like the direction this is going. Here's a few thoughts that I could only after looking at your PR:

The check `*/*/*` doesn't seem to be the best anymore because it's including at least two cases and it's becoming a little difficult to follow. It would be cool to list in the documentation all accepted patterns for specifying packages, then refactor the logic to better account for them. What I can come up with from the top of my head:

- Fully qualified
  - `https://github.com/basherpm/basher.git` (`-ssh` flag can't be present)
  - `ssh://git@github.com/basherpm/basher.git` (`-ssh` flag can't be present)

- Git url
  - `git@github.com:basherpm/basher.git` (`-ssh` flag can't be present)

- Shortcuts
  - `basherpm/basher` with or without `--ssh`
  - `github.com/basherpm/basher` with or without `--ssh`

It also seems that multiple level packages are only supported when specifying fully qualified URLs (which might be a good restriction), or we could change the second shortcut to be `github.com:basherpm/basher`, which kind of resembles the git url format. I think having support for multiple level only when using full URLs is better for now because it's simpler and we could still make other changes in the future.

`basher install` is missing new tests.

Besides all that, you also need to make sure you're able to run any commands which take a package name only (update, uninstall etc) and add the corresponding tests. Updating the documentation describing how to specify a `package` is also important in the README.

**Comment by lordvlad** [comment]
Thanks for the review, @juanibiapina. I'll see if I find some time this weekend to improve on the PR

**Comment by juanibiapina** [comment]
closing due to inactivity.

**Pull request closed without merging**